### ISSUE TITLE
Panics are getting hard to understand

### ISSUE THREAD

**Issue by mockersf** [bevyengine/bevy#8215]
## Bevy version

current main (4f16d6e0dc14c12ab29b72880d5d3d22003eed72)

## What you did

```rs
use bevy::prelude::*;

fn main() {
    App::new()
        .add_plugins(DefaultPlugins)
        .add_systems(Update, panicking_system)
        .run();
}

fn panicking_system() {
    panic!("oooh scary");
}
```

## What went wrong

It's hard to understand which panic I should care about. Well not quite in this specific case as I put an explicit message, but it usually is. It can be even worse if the panic is from inside Bevy code.

```
   Compiling panicking v0.1.0
    Finished release [optimized] target(s) in 2.04s
     Running `target/release/panicking`
2023-03-25T22:50:20.189287Z  INFO bevy_render::renderer: AdapterInfo { name: "Apple M1 Max", vendor: 0, device: 0, device_type: IntegratedGpu, driver: "", driver_info: "", backend: Metal }
2023-03-25T22:50:20.317469Z  INFO bevy_winit::system: Creating new window "Bevy App" (0v0)
2023-03-25T22:50:20.392074Z  INFO bevy_diagnostic::system_information_diagnostics_plugin::internal: SystemInfo { os: "MacOS 13.1 ", kernel: "22.2.0", cpu: "Apple M1 Max", core_count: "10", memory: "64.0 GiB" }
thread 'Compute Task Pool (3)' panicked at 'oooh scary', main.rs:14:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
thread 'Compute Task Pool (3)' panicked at 'A system has panicked so the executor cannot continue.: RecvError', crates/bevy_ecs/src/schedule/executor/multi_threaded.rs:194:60
thread 'main' panicked at 'called `Option::unwrap()` on a `None` value', bevy/crates/bevy_tasks/src/task_pool.rs:376:49
thread 'Compute Task Pool (3)' panicked at 'called `Result::unwrap()` on an `Err` value: RecvError', crates/bevy_render/src/pipelined_rendering.rs:110:77
thread '<unnamed>' panicked at 'called `Option::unwrap()` on a `None` value', bevy/crates/bevy_tasks/src/task_pool.rs:376:49
```

**Comment by alice-i-cecile** [comment]
Yeah, the panic messages are significantly degraded in 0.10 following the new approach to tasks and scheduling.

**Comment by nicopap** [comment]
It is also a lot harder to decrypt backtraces with `RUST_BACKTRACE=1` since you have to figure out which one is relevant.

**Comment by rparrett** [comment]
Related? #8152

**Comment by JonahPlusPlus** [comment]
Made a simple PR that fixes this issue: #8219. It seems that fixing the panic in the multi-threaded scheduler also fixes the other panic messages.
